gravitee-rest-api-portal/gravitee-rest-api-portal-rest/src/test/java/io/gravitee/rest/api/portal/rest/resource/CategoriesResourceTest.java

```diff
@@ -125,4 +130,51 @@ public void shouldGetNoPublishedApiAndNoLink() {
         assertNull(links);
 
     }
+
+    @Test
+    public void shouldGetNothingIfAllCategoriesEmpty() {
+        // modify mocks: 0 APIs returned for user in any categories
+        Set<ApiEntity> emptyMockApis = new HashSet<>();
+        doReturn(emptyMockApis).when(apiService).findPublishedByUser(any());
+        doReturn(0L).when(categoryService).getTotalApisByCategory(any(), any()) ;
+
+        final Response response = target().request().get();
+        assertEquals(HttpStatusCode.OK_200, response.getStatus());
+
+        Mockito.verify(apiService).findPublishedByUser(any());
+
+        CategoriesResponse categoriesResponse = response.readEntity(CategoriesResponse.class);
+
+        assertEquals(0, categoriesResponse.getData().size());
+    }
+
+    @Test
+    public void shouldGetOnlyNonEmptyCategories() {
+        Set<ApiEntity> mockApis = new HashSet<>();
+        doReturn(mockApis).when(apiService).findPublishedByUser(any());
+
+        //mock configuration for categories: C1 has 1 API for user, C2 has 0, C3 (hidden anyway) has 0
+        existingCategories.stream()
```

Comment:
To simplify, I would have write this mock in the @Before method. It would not have any impact on the previous test since, `getTotalApisByCategory` mock is overriden